ChromeExtension/js/GeniusLinkServiceClient.js

```diff
@@ -0,0 +1,54 @@
+GeniusLinkServiceClient.prototype = JsvServiceClient.prototype;
+
+
+function GeniusLinkServiceClient(baseUrl, apiKey, apiSecret) {
+    var that = this;
+    JsvServiceClient.call(this, baseUrl);
+    that.baseSyncReplyUri = baseUrl;
+    that.baseAsyncOneWayUri = baseUrl;
+    that.apiSecret = apiSecret;
+    that.apiKey = apiKey;
+    that.getFromService = function (url, objectToProcess, callback,error) {
+        objectToProcess.apiKey = that.apiKey;
+        objectToProcess.apiSecret = that.apiSecret;
+        objectToProcess.format = "jsv";
+
+        GeniusLinkServiceClient.prototype.getFromService.call(this, url, objectToProcess, callback, error);
+    }
+
+
+
+
+}
+GeniusLinkServiceClientPost.prototype = JsvServiceClient.prototype;
```

Comment:
So this is wrong all you need to do is add the method to the original object not create one new object for each method.

This should look like

`function GeniusLinkServiceClient(baseUrl, apiKey, apiSecret) {
 +    var that = this;
 +    JsvServiceClient.call(this, baseUrl);
 +    that.baseSyncReplyUri = baseUrl;
 +    that.baseAsyncOneWayUri = baseUrl;
 +    that.apiSecret = apiSecret;
 +    that.apiKey = apiKey;
 +    that.getFromService = function (url, objectToProcess, callback,error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +        objectToProcess.format = "jsv";
 +
 +        GeniusLinkServiceClient.prototype.getFromService.call(this, url, objectToProcess, callback, error);
 +    };
 +
 + that.postFormDataToService = function (url, objectToProcess, callback, error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +
 +        GeniusLinkServiceClient.prototype.postFormDataToService.call(this, url, objectToProcess,
 +            callback, error);
 +    };
 +
that.postToService = function (url, objectToProcess, callback, error) {
 +        objectToProcess.apiKey = that.apiKey;
 +        objectToProcess.apiSecret = that.apiSecret;
 +
 +        GeniusLinkServiceClient.prototype.postFormDataToService.call(this, url, objectToProcess,
 +            callback, error);
 +    };
 +
 +}`